File: Java/src/covariance/maths/CovVertex.java
package covariance.maths;

import covariance.abs.AbsGraph;
import covariance.abs.AbsVertex;

public class CovVertex extends AbsVertex
{
    public CovVertex(Integer label)
    {
        super(label);
    }

    public void setGraph(CovGraph g)
    {
        super.setGraph(g);        
    }
    
    @Override
    public void setGraph(AbsGraph g)
    {
        try
        {
            setGraph((CovGraph) g);
        }
        catch (Exception e)
        {
            e.printStackTrace();
        }
    }
}

File: scala/src/covariance/maths/CovVertex.scala
package covariance.maths

import covariance.abs.AbsGraph
import covariance.abs.AbsVertex

class CovVertex(label: Integer) extends AbsVertex(label) {

  def setGraph(g: CovGraph): Unit = {
    super.setGraph(g)
  }

  override def setGraph(g: AbsGraph): Unit = {
    try {
      setGraph(g.asInstanceOf[CovGraph])
    } catch {
      case e: Exception =>
        e.printStackTrace()

    }
  }

}
